server/src/packages/rwcds-form/rwcds-form.service.ts

```diff
@@ -140,25 +140,26 @@ export class RwcdsFormService {
     const zrTypes = affectedZoningResolutions.map(zr => zr['dcp_zoningresolutiontype@OData.Community.Display.V1.FormattedValue']);
     const { records: projectActions } = await this.crmService.get(`dcp_projectactions`, `
       $filter=_dcp_project_value eq ${_dcp_projectid_value}
+      &$expand=dcp_ZoningResolution
     `);
 
     await Promise.all(projectActions.map(action => {
       const projectActionLabel = action['_dcp_action_value@OData.Community.Display.V1.FormattedValue'];
 
       // Lookup: dcp_affectedzoningresolutions asks for a ZR type,
-      const [actionLabel, actionCode] = ZONING_RESOLUTION_TYPES.find((label) => label === projectActionLabel) || ['ZA', 717170000];
+      const { label, code } = ZONING_RESOLUTION_TYPES.filter((zr) => zr.label === projectActionLabel)[0] || { label: '', code: null };
 
-      if (!actionLabel) console.log(`Could not find Affected ZR Type for ${projectActionLabel}`);
+      if (!label) console.log(`Could not find Affected ZR Type for ${projectActionLabel}`);
 
-      if (!zrTypes.includes(actionLabel)) {
+      if (!zrTypes.includes(label) && label) {
         return this.crmService.create(`dcp_affectedzoningresolutions`, {
-          'dcp_zoningresolutiontype': actionCode, // this is a coded value
-          'dcp_zrsectionnumber': action.dcp_zrsectionnumber,
+          'dcp_zoningresolutiontype': code, // this is a coded value
+          'dcp_zrsectionnumber': action.dcp_ZoningResolution ? action.dcp_ZoningResolution.dcp_zoningresolution : null,
           'dcp_modifiedzrsectionnumber': action.dcp_zrmodifyingzrtxt,
           'dcp_rwcdsform@odata.bind': `/dcp_pasforms(${dcp_rwcdsformid})`,
         });
-      } else {
-        console.log(`zoning resolution with id ${projectActionLabel} already exists`);
+      } else if (zrTypes.includes(label) && label) {
+        console.log(`zoning resolution with id ${label} already exists`);
```

Comment:
small nit: I've been noticing that this keeps logging out to the terminal. Would prefer removing this console log here

Reply:
will do!